pkg/container-utils/docker/docker.go

```diff
@@ -235,6 +236,25 @@ func (c *DockerClient) Close() error {
 	return nil
 }
 
+// Gets the image digest for the given image ID, if the digest exists.
+// The digest is usually only available if the image was either pulled from a registry, or if the image was pushed to a registry, which is when the manifest is generated and its digest calculated.
+// Note: This function only works for already running containers and not for containers that are being created.
+func (c *DockerClient) getContainerImageDigest(imageId string) string {
+	imageInspect, _, err := c.client.ImageInspectWithRaw(context.Background(), imageId)
+	if err != nil {
+		log.Warnf("Failed to get image digest for image %s: %s", imageId, err)
+		return ""
+	}
+
+	if len(imageInspect.RepoDigests) == 0 {
+		log.Warnf("No digest found for image %s", imageId)
+		return ""
+	}
+
+	imageWithDigest := imageInspect.RepoDigests[0]
+	return strings.Split(imageWithDigest, "@")[1]
```

Comment:
You can test `len(strings.Split(...))` to avoid a panic if the string does not container "@".